src/Mvc/Mvc.Core/src/FromRouteAttribute.cs

```diff
@@ -10,12 +12,14 @@ namespace Microsoft.AspNetCore.Mvc
     /// Specifies that a parameter or property should be bound using route-data from the current request.
     /// </summary>
     [AttributeUsage(AttributeTargets.Parameter | AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
-    public class FromRouteAttribute : Attribute, IBindingSourceMetadata, IModelNameProvider
+    public class FromRouteAttribute : Attribute, IBindingSourceMetadata, IModelNameProvider, IFromRouteAttribute
```

Comment:
If we wanted to avoid introducing a new set of interfaces, we should be able to get away by moving `IBindingSourceMetadata`, `IModelNameProvider` and `BindingSource` up stack.

Reply:
Personally, I think we should just type forward all the attributes and the small number of types needed to support the interfaces they implement. @davidfowl wants to avoid any dependency on the MVC namespace though so we can start from a clean slate. Maybe we can wear him down though 😄

Reply:
Right, I'm willing to move those into the core with a horrible namespace *after* we get some core functionality working without them. I want to pull things in as required.

Reply:
Nah

Reply:
> 'm willing to move those into the core with a horrible namespace after we get some core functionality working without them

Is there some way we should be tracking this? It would be easy to end up with 💩 types that don't truly need to exist otherwise.

Reply:
ICanHasViewContext 🙃. Sure we should track it somehow. I also think we should avoid API review until we've gone through some iterations